app/signup/page.tsx

```diff
@@ -0,0 +1,82 @@
+"use client";
+import { useRef, useState } from "react";
+import { useRouter } from "next/navigation";
+import Link from "next/link";
+import { register } from "@/actions/register";
+export default function Register() {
```

Comment:
Why this page is not accessible by navigating directly with URL?

Reply:
Because I added a feature to the provider that prevents access to the registration and login page if you are already logged in.